src/libraries/System.Collections.Immutable/tests/Frozen/FrozenDictionaryTests.cs

```diff
@@ -421,6 +421,17 @@ public abstract class FrozenDictionary_Generic_Tests_base_for_numbers<T> : Froze
         protected override T CreateTValue(int seed) => CreateTKey(seed);
 
         protected abstract T Next(Random random);
+
+        protected static long NextLong(Random random)
+        {
+#if NET6_0_OR_GREATER
+            return random.NextInt64(long.MinValue, long.MaxValue);
+#else
+            byte[] bytes = new byte[8];
+            random.NextBytes(bytes);
+            return BitConverter.ToInt64(bytes, 0);
+#endif
```

Comment:
It's not going to matter, but technically these aren't the same thing.  NextInt64(long.MinValue, long.MaxValue) will never produce long.MaxValue, but the else code can.

Reply:
> will never produce long.MaxValue, but the else code can. 

Good point! After thinking about it I decided to remove the NET6_0_OR_GREATER case as it could cause another kind of issues: generate different output for the same input and hence get us to a situation where exactly the same test is passing for one moniker and failing for another.